blockchain/validate_test.go

```diff
@@ -313,66 +315,69 @@ func TestSigOpsLimitsWithMultiMBBlocks(t *testing.T) {
 	// tx1:      1 sigOps    <=1MB
 	// tx2:      1 sigOps    <=1MB
 	// block:    3 sigOps    < 1MB
-
-	// Add many NOOPs to the coinbase out script which can be arbitrary.
-	// This gets the block >1MB without breaking Tx size limits
-	bigEnoughScript := append(baseScriptForSize, repeatScript(979800, txscript.OP_NOP)...)
+	// Add 3 * 400k NOOPs to get >1MB block without breaking Tx size limits
+	manyNoOps := repeatScript(400000, txscript.OP_NOP)
+	for _, tx := range baseBlock.Transactions {
+		tx.TxOut[0].PkScript = append(tx.TxOut[0].PkScript, manyNoOps...)
+	}
 
 	// 1. Pass at per-Tx sigOps limit
 	// coinbase:     20k sigOps (ok)     <=1MB (ok)
 	// tx1:            1 sigOps (ok)     <=1MB (ok)
 	// tx2:            1 sigOps (ok)     <=1MB (ok)
 	// block:    20k + 2 sigOps (ok)     > 1MB (ok)
-	baseBlock.Transactions[0].TxOut[0].PkScript = bigEnoughScript
 	atTxLimit, err := newTestBlock(baseBlock, tip, 20000, 1, 1)
 	if err != nil {
 		t.Fatalf("Unexpected error creating sigOps test block: %v", err)
 	}
-	err = attachBigBlock(atTxLimit)
+	err = validateBigBlock(atTxLimit)
 	if err != nil {
-		t.Fatalf("Expected to validate but got: %v", err)
+		if rule, ok := err.(RuleError); ok {
+			if rule.ErrorCode == ErrTooManySigOps {
+				t.Fatalf("Expected to validate but got: %v", err)
+			}
+		}
+
```

Comment:
I don't see the value of silencing other errors here? If the attach fails for whatever reason, don't we want to know about it? Because the rest of the test depends on this block attaching correctly? Sorry I don't know the conditions when an error would be ok vs fatal regarding these tests.

Reply:
Nevermind about this. As long as we get the tests we need (and I hope explicitly state that we are ignoring errors), then the test does its job.